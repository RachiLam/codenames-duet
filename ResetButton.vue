<template>
  <button @click="reset" />
</template>

<script>
export default {
  methods: {
    reset() {
      window.location.reload();
    },
  },
};
</script>

<style lang="scss" scoped>
  @import './variables';

  button {
    width: $reset-button-size;
    height: $reset-button-size;
    background-color: $accent-light;
    border-radius: 25%;
  }
</style>
