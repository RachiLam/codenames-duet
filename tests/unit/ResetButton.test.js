import { shallowMount } from '@vue/test-utils';
import ResetButton from '../../src/components/ResetButton.vue';

describe('components/ResetButton', () => {
  describe('when not activated', () => {
    test('renders as inactive', () => {
      const wrapper = shallowMount(ResetButton, {
        propsData: {
          activated: false,
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('when activated', () => {
    test('renders as activated', () => {
      const wrapper = shallowMount(ResetButton, {
        propsData: {
          activated: true,
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
