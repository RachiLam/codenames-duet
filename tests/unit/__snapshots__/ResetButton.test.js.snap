// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ResetButton when activated renders as activated 1`] = `
<button
  class="activated"
/>
`;

exports[`components/ResetButton when not activated renders as inactive 1`] = `
<button
  class=""
/>
`;
